File: CPP_Joc_Windows_Android/SH_Client_Win_Cpp_Cmake/App/src/base/crypto/cipher/EncryptCipher.cpp
#include <base/encoding/base64/Base64.h>
#include "EncryptCipher.h"
#include <base/MM.h>

using namespace base::crypto;

EncryptCipher::EncryptCipher(std::shared_ptr<IAlgorithm> algorithm)
    : super(algorithm)
{
    //void
}

std::string EncryptCipher::encryptStringToBase64(std::string& data) {
    char* eData = new char[data.size()];

    processBytes(data.data(), &eData[0], static_cast<int>(data.size()));

    std::string base64String = base::encoding::Base64::encode(&eData[0], data.size());

    delete[] eData;

    return base64String;
}

EncryptCipher::~EncryptCipher() {
    //void
}

File: NodeJs_TypeScript_JavaScript_Server/Project_NodeJs_VSCode/src/base/crypto/cipher/EncryptCipher.ts
import { BasicCipher } from "./BasicCipher";
import { IEncryptCipher } from "./IEncryptCipher";
import { Base64 } from "base/encoding/base64/Base64";
import { IAlgorithm } from "../algorithm/IAlgorithm";

export class EncryptCipher extends BasicCipher implements IEncryptCipher {

    public constructor(algorithm: IAlgorithm) {
        super(algorithm);

        //void
    }

    public encryptIntToBuffer(data: number): Buffer {
        let bData: Buffer = Buffer.allocUnsafe(4);
        bData.writeInt32BE(data, 0);

        this.processBytes(
            bData.buffer, bData.byteOffset,
            bData.buffer, bData.byteOffset,
            bData.byteLength
        );

        return bData;
    }

    public encryptStringToBase64(data: string): string {
        let bData: Buffer = Buffer.from(data, "utf8");

        this.processBytes(
            bData.buffer, bData.byteOffset,
            bData.buffer, bData.byteOffset,
            bData.byteLength
        );

        let ret: string = Base64.encode(bData);

        return ret;
    }

}
